internal/metrics/metrics.go

```diff
@@ -14,3 +16,103 @@ func Register(serveMux *http.ServeMux) {
 		}),
 	))
 }
+
+const (
+	namespace           = "strong_duckling"
+	subSystemTcpChecker = "tcp_checker"
+)
+
+const (
+	strongswanVersion     = "strongswan_version"
+	strongDucklingVersion = "version"
+)
+
+type PrometheusReporter struct {
+	registry   prometheus.Registerer
+	version    *prometheus.GaugeVec
+	TcpChecker *TcpChecker
+}
+
+type TcpChecker struct {
+	open             *prometheus.GaugeVec
+	connectedTotal   *prometheus.CounterVec
+	disconectedTotal *prometheus.CounterVec
+
+	previousOpenState *bool
+}
+
+func NewPrometheusReporter(reg prometheus.Registerer) (*PrometheusReporter, error) {
+	r := PrometheusReporter{
+		registry: reg,
+		version: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      "info",
+			Help:      "Version info of strong_duckling",
+		}, []string{strongswanVersion, strongDucklingVersion}),
+		TcpChecker: &TcpChecker{
+			open: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: namespace,
+				Subsystem: subSystemTcpChecker,
+				Name:      "open_info",
+				Help:      "Is TCP open is 1 otherwise 0",
```

Comment:
Usually prom help texts are more like a sentence.

```suggestion
				Help:      "Indicates if TCP address:port is open with 1 and closed with 0",
```

I think by convention it is expected that 1 is open and 0 is not, but that is up to you to decide :)